src/Provider/Yahoo.php

```diff
@@ -120,11 +120,11 @@ public function getUserProfile()
             $userProfile->gender = 'male';
         }
 
-        // I ain't getting no emails on my tests. go figures..
+        // E-mail is returned only with sdpp-w scope ( Read/Write (Public and Private) )
         foreach ($data->filter('emails')->toArray() as $item) {
-            if ($data->get('primary')) {
-                $userProfile->email         = $data->get('handle');
-                $userProfile->emailVerified = $data->get('handle');
+            if ($item->primary) {
```

Comment:
let's use the following:

```php
$item = new Data\Collection($item);

if ($item->get('primary')) {
    $userProfile->email = $item->get('handle');
    $userProfile->emailVerified = $item->get('handle');
}
```

Reply:
That is correct, i made the same change in my project but i forgot to add it here.

Reply:
@antiqque hey, any updates?

Reply:
I'm sorry but this is my first time doing this and i'm not sure what else i'm supposed to do.